Enable 50G/100G/200G ethtool link modes for XS8 kernel

This is a patch specifically for NetScaler's Everest platform.

The standard 50G/100G/200G link modes do not fit within the 4.19 kernel ABI so
we had to drop 3 of them to squeeze the rest into the available space.

The three missing link modes are:
  ETHTOOL_LINK_MODE_50000baseLR_ER_FR_Full_BIT
  ETHTOOL_LINK_MODE_100000baseLR2_ER2_FR2_Full_BIT
  ETHTOOL_LINK_MODE_200000baseLR4_ER4_FR4_Full_BIT

This patch requires matching changes in the kernel and ethtool userspace tool.

index bae4832..22a3a2a 100644
--- a/drivers/net/ethernet/mellanox/mlx5/core/en_ethtool.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/en_ethtool.c
@@ -184,7 +184,6 @@ void mlx5e_build_ptys2ethtool_map(void)
 				       ETHTOOL_LINK_MODE_50000baseKR_Full_BIT,
 				       ETHTOOL_LINK_MODE_50000baseSR_Full_BIT,
 				       ETHTOOL_LINK_MODE_50000baseCR_Full_BIT,
-				       ETHTOOL_LINK_MODE_50000baseLR_ER_FR_Full_BIT,
 				       ETHTOOL_LINK_MODE_50000baseDR_Full_BIT);
 	MLX5_BUILD_PTYS2ETHTOOL_CONFIG(MLX5E_CAUI_4_100GBASE_CR4_KR4, ext,
 				       ETHTOOL_LINK_MODE_100000baseKR4_Full_BIT,
@@ -195,12 +194,10 @@ void mlx5e_build_ptys2ethtool_map(void)
 				       ETHTOOL_LINK_MODE_100000baseKR2_Full_BIT,
 				       ETHTOOL_LINK_MODE_100000baseSR2_Full_BIT,
 				       ETHTOOL_LINK_MODE_100000baseCR2_Full_BIT,
-				       ETHTOOL_LINK_MODE_100000baseLR2_ER2_FR2_Full_BIT,
 				       ETHTOOL_LINK_MODE_100000baseDR2_Full_BIT);
 	MLX5_BUILD_PTYS2ETHTOOL_CONFIG(MLX5E_200GAUI_4_200GBASE_CR4_KR4, ext,
 				       ETHTOOL_LINK_MODE_200000baseKR4_Full_BIT,
 				       ETHTOOL_LINK_MODE_200000baseSR4_Full_BIT,
-				       ETHTOOL_LINK_MODE_200000baseLR4_ER4_FR4_Full_BIT,
 				       ETHTOOL_LINK_MODE_200000baseDR4_Full_BIT,
 				       ETHTOOL_LINK_MODE_200000baseCR4_Full_BIT);
 	MLX5_BUILD_PTYS2ETHTOOL_CONFIG(MLX5E_100GAUI_1_100GBASE_CR_KR, ext,
